### ISSUE TITLE
[FEA] cuDF does not support `DataFrame` initialization with `numpy array`

### ISSUE THREAD

**Issue by vivanov10** [rapidsai/cudf#5545]
**Is your feature request related to a problem? Please describe.**

cuDF does not support `DataFrame` initialization with `numpy array`

**Describe the solution you'd like**

Implement initialization with `numpy array` as in Pandas

**Describe alternatives you've considered**

Convert `numpy array` to list of lists, or a dictionary of lists first

**Additional context**

Pandas example:
```python
import pandas as pd
import numpy as np
data = pd.DataFrame(data=np.array([[1, 10, 100, 1000, 10000], [2, 20, 200, 2000, 20000]]), columns=['a', 'b', 'c', 'd', 'e'])
```

cuDF example:
```python
import cudf as pd
import numpy as np
data = pd.DataFrame(
data=np.array([[1, 10, 100, 1000, 10000], [2, 20, 200, 2000, 20000]]), columns=['a', 'b', 'c', 'd', 'e'])
# TypeError: data must be list or dict-like
```

Traceback:
```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-29-ccb0dbfc6713> in <module>
      3 import numpy as np
      4 data = pd.DataFrame(
----> 5 data=np.array([[1, 10, 100, 1000, 10000], [2, 20, 200, 2000, 20000]]), columns=['a', 'b', 'c', 'd', 'e'])
      6 # TypeError: data must be list or dict-like

~/miniconda3/envs/rapids14/lib/python3.7/contextlib.py in inner(*args, **kwds)
     72         def inner(*args, **kwds):
     73             with self._recreate_cm():
---> 74                 return func(*args, **kwds)
     75         return inner
     76 

~/miniconda3/envs/rapids14/lib/python3.7/site-packages/cudf/core/dataframe.py in __init__(self, data, index, columns, dtype)
    226             else:
    227                 if not is_dict_like(data):
--> 228                     raise TypeError("data must be list or dict-like")
    229 
    230                 self._init_from_dict_like(data, index=index, columns=columns)

TypeError: data must be list or dict-like

```

**Comment by brandon-b-miller** [comment]
Hi @vivanov10 , what version of cuDF are you using here? Using the recent 0.15 this seems to work.

```
>>> import numpy as np
>>> import cudf
>>> cudf.DataFrame(data=np.array([[1, 10, 100, 1000, 10000], [2, 20, 200, 2000, 20000]]), columns=['a', 'b', 'c', 'd', 'e'])
   a   b    c     d      e
0  1  10  100  1000  10000
1  2  20  200  2000  20000
>>> cudf.__version__
'0.15.0a+1298.g636a8a0e4'
```

**Comment by kkraus14** [comment]
This was resolved by #5305 in 0.15.

**Comment by vivanov10** [comment]
Hi @kkraus14 , I use 0.14 stable version.

**Comment by kkraus14** [comment]
> Hi @kkraus14 , @brandon-b-miller , I use 0.14 stable version.

Understood. Issues are closed as they are fixed in the current development version. This fix will be included in the 0.15 release.

**Comment by vivanov10** [comment]
That's great, thanks :)
